stats/stats-server/src/config/json_config.rs

```diff
@@ -0,0 +1,80 @@
+use super::{
+    chart_info::{CounterInfo, LineChartInfo},
+    toml_config,
+};
+use convert_case::{Case, Casing};
+use serde::Deserialize;
+use std::collections::BTreeMap;
+
+#[derive(Default, Debug, Clone, Deserialize)]
+#[serde(default, deny_unknown_fields)]
+pub struct LineChartSection {
+    pub title: String,
+    pub charts: BTreeMap<String, LineChartInfo>,
+}
+
+#[derive(Default, Debug, Clone, Deserialize)]
+#[serde(default, deny_unknown_fields)]
+pub struct Config {
+    pub counters: BTreeMap<String, CounterInfo>,
+    pub lines: BTreeMap<String, LineChartSection>,
+}
+
+impl From<Config> for toml_config::Config {
+    fn from(value: Config) -> Self {
+        Self {
+            counters: value
+                .counters
+                .into_iter()
+                .map(toml_config::CounterInfo::from)
+                .collect(),
+            lines: value
+                .lines
+                .into_iter()
+                .map(toml_config::LineChartSection::from)
+                .collect::<Vec<_>>()
+                .into(),
+        }
+    }
+}
+
+impl From<Vec<toml_config::LineChartSection>> for toml_config::LineCharts {
+    fn from(sections: Vec<toml_config::LineChartSection>) -> Self {
+        Self { sections }
+    }
+}
+
+impl From<(String, LineChartSection)> for toml_config::LineChartSection {
+    fn from((id, section): (String, LineChartSection)) -> Self {
+        Self {
+            id,
+            title: section.title,
+            charts: section
+                .charts
+                .into_iter()
+                .map(toml_config::LineChartInfo::from)
+                .collect(),
+        }
+    }
+}
+
+impl From<(String, LineChartInfo)> for toml_config::LineChartInfo {
+    fn from((id, info): (String, LineChartInfo)) -> Self {
+        Self {
+            id: id.to_case(Case::Camel),
```

Comment:
I suggest to add a from statement as well, just to make sure the library recognizes the initial cases correctly and do not recognize any other pattern

```suggestion
            id: id.from_case(Case::Snake).to_case(Case::Camel),
```